### PR TITLE
Update with Gateway API

### PR DESCRIPTION
**Pull request by kevinch**
#### What is the purpose of this pull request?
Update the Credit Control extension point at Checkout-Ui.

#### How should this be manually tested?
`link`, add product to cart on store front and add the workspace like `https://ambienteqa.myvtex.com/checkout/?workspace=kevin#/payment`

#### Screenshots or example usage
screen shot 2018-08-23 at 16 43 07

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: extension_point_checkout/react/components/EnoughCredit.js [hunk 1]
```diff
@@ -33,51 +39,43 @@ class EnoughCredit extends PureComponent {
...
-                    ),
-                )}
+                            })}`
+                        : intl.formatNumber(paymentValue, {
+                            style: 'currency',
+                            currency: currencyCode,
+                          })}{' '}
+                      {intl.formatMessage({ id: 'in' })}{' '}
+                      {installment.paymentOptions.map((op, i) => {
+                        let r = ''
```
Comment (nandoacoelho) [inline extension_point_checkout/react/components/EnoughCredit.js]: `r` could be a little more declarative

File: extension_point_checkout/react/locales/pt-BR.json [hunk 2]
```diff
@@ -13,8 +11,13 @@
   "withInterests": "{installments}x de {value} com juros de {interest}%",
   "withoutInterests": "{installments}x de {value} sem juros",
 
-  "error_0000": "Ocorreu um erro.",
-  "error_401": "Sem autorização",
-  "error_2003": "Conta não encontrada para {clientId}",
-  "error_4001": "Acesso negado."
+  "verifyAccount_400": "Erro ao buscar sua conta.",
+  "verifyAccount_404": "Conta não encontrada.",
+  "verifyCheckingAccount_404": "Nenhum Checking Account encontrado para seu email.",
```
Comment (akaFTS) [inline extension_point_checkout/react/locales/pt-BR.json]: Nenhuma conta corrente encontrada

File: extension_point_checkout/react/components/CheckoutListener.js [hunk 3]
```diff
@@ -47,18 +51,36 @@ class CheckoutListener extends Component {
...
+          } else {
+            this.fetchAccountById(data)
+          }
+        })
+        .catch(error => {
+          this.manageError({
+            response: error.response,
+            step: 'verifyCheckingAccount',
+          })
+        })
```
Comment (akaFTS) [inline extension_point_checkout/react/components/CheckoutListener.js]: `async/await/throw` would make this function look much nicer

File: extension_point_checkout/react/components/EnoughCredit.js [hunk 4]
```diff
@@ -33,51 +39,43 @@ class EnoughCredit extends PureComponent {
...
+                          r = ', '
+                        }
+
+                        if (i === optionsLength - 2 && optionsLength > 1) {
+                          r = ` ${intl.formatMessage({ id: 'and' })} `
+                        }
+
+                        return `${op.days}${r}`
+                      })}
+                      {` ${intl.formatMessage({ id: 'days' })} `}
```
Comment (akaFTS) [inline extension_point_checkout/react/components/EnoughCredit.js]: This is pretty hard to read, maybe an approach with `join` or `reduce` would make it look better?

File: extension_point_checkout/react/components/EnoughCredit.js (hunk 1 above)
Comment (kevinch) [reply extension_point_checkout/react/components/EnoughCredit.js]: true

File: extension_point_checkout/react/locales/pt-BR.json (hunk 2 above)
Comment (kevinch) [reply extension_point_checkout/react/locales/pt-BR.json]: Nope, é um conceito diferente de uma conta corrente de banco.

File: extension_point_checkout/react/components/CheckoutListener.js [hunk 5]
```diff
@@ -72,7 +72,7 @@ class CheckoutListener extends Component {
               response: { data: { code: 404 } },
               step: 'verifyCheckingAccount',
             })
-          : this.fetchAccountById(data)
+          : this.fetchDeadlines(data.data[0])
```
Comment (nandoacoelho) [inline extension_point_checkout/react/components/CheckoutListener.js]: Did you check if this exists before referencing?

Comment (kevinch) [reply extension_point_checkout/react/components/CheckoutListener.js]: yes, right above `isEmpty(data.data[0])`

**Pull request merged**